docs/explainer.md

```diff
@@ -1,87 +1,91 @@
+1. **[WPACK](https://github.com/WICG/webpackage):** It is intended to pack a collection of web resources in terms of HTTP exchanges (including HTTP requests, responses, signatures, and other metadata) so that a browser can load them locally and recover the web transaction states securely without accessing the origin. To achieve this, an HTTP-specific signature mechanism is defined, which is apparently not suitable for MiniApp, which does not rely on HTTP exchanges. Moreover, WPACK proposes to define a new CBOR-based archive format instead of using ZIP for web-specific [reasons](https://github.com/WICG/webpackage/issues/45) that do not apply to MiniApp. For instance, ZIP resources are file-based, so they are not efficient for HTTP response headers in WPACK but can serve MiniApp very well and are already used widely in mobile application packaging implementations (e.g., Android APK, QuickApp, MiniProgram). 
+2. **[Web Package (obsolete)](https://www.w3.org/TR/2015/WD-web-packaging-20150115/):** It is an unfinished work as the ancestor of WPACK. It chooses not to use ZIP but defines a new Streamable Package Format for specific reasons that are not seen by MiniApp. Again, it is designed as closely coupled with HTTP (headers) which is not suitable for MiniApp.
+3. **[Widgets (obsolete)](https://www.w3.org/TR/widgets):** It is an old implementation of Packaged Web Apps called Widgets. It is ZIP-based but has specific file/folder structure and naming constraints, and it uses XML-based configuration documents. These constraints are not compatible with MiniApp. In addition, it uses [XMLDSIG](http://www.w3.org/TR/xmldsig-core1/) for the digital signature, which can only protect the signed files but not the entire package file itself. One can add/remove signature files, add other files or blocks into the ZIP package, or alter the ZIP metadata without being noticed. Such vulnerability is not acceptable by MiniApp.
+4. **[LPF](https://www.w3.org/TR/lpf/):** It is a lightweight ZIP-based packaging format that contains digital publication materials. It mandates publication-specific files (i.e., `publication.json`, `index.html`) that are not needed by a MiniApp, while misses MiniApp specific files like `manifest.json` and `app.js`. Also, it does not cover the security concerns, such as the integrity protection of the whole package that a MiniApp addresses.
 
-A detailed comparison between MiniApp package and other packaging technologies is given in the following table:
+The following table shows a detailed comparison between a MiniApp package and other packaging technologies
```

Comment:
```suggestion
The following table shows a detailed comparison between a MiniApp package and other packaging technologies:
```

This is not the caption/title of the table, but a sentence, so punctuation is needed at the end of the sentence.